### ISSUE TITLE
enhancement: a11y rule "no-noninteractive-element-to-interactive-role" fails with a warning.

### ISSUE THREAD

**Issue by FatumaA** [open-sauced/app#4129]
### Describe the bug

[jsx-a11y eslint plugin](https://www.npmjs.com/package/eslint-plugin-jsx-a11y) was recently added to the project and this rule -
jsx-a11y/no-noninteractive-element-to-interactive-role - fails with a warning.

This issue offers more context: https://github.com/open-sauced/app/issues/2264

### Steps to reproduce

The following components and files are causing the rule to fail:

1. components/TabList/tab-list.tsx
2. components/organisms/ToolList/nav.tsx

**Comment by nickytonline** [comment]
.take

**Comment by Mandeep56Singh** [comment]
.take
